plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/BaseMariaDbConnectorTest.java

```diff
@@ -262,6 +262,32 @@ public void testDeleteWithLike()
                 .hasStackTraceContaining("TrinoException: Unsupported delete");
     }
 
+    // Overridden because the method from BaseConnectorTest fails on one of the assertions, see TODO below
+    @Test
+    @Override
+    public void testInsertIntoNotNullColumn()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "insert_not_null", "(nullable_col INTEGER, not_null_col INTEGER NOT NULL)")) {
+            assertUpdate(format("INSERT INTO %s (not_null_col) VALUES (2)", table.getName()), 1);
+            assertQuery("SELECT * FROM " + table.getName(), "VALUES (NULL, 2)");
+            assertQueryFails(format("INSERT INTO %s (nullable_col) VALUES (1)", table.getName()), errorMessageForInsertIntoNotNullColumn("not_null_col"));
+            assertQueryFails(format("INSERT INTO %s (not_null_col, nullable_col) VALUES (NULL, 3)", table.getName()), "NULL value not allowed for NOT NULL column: not_null_col");
+            assertQueryFails(format("INSERT INTO %s (not_null_col, nullable_col) VALUES (TRY(5/0), 4)", table.getName()), "NULL value not allowed for NOT NULL column: not_null_col");
+            assertQueryFails(format("INSERT INTO %s (not_null_col) VALUES (TRY(6/0))", table.getName()), "NULL value not allowed for NOT NULL column: not_null_col");
+            assertQueryFails(format("INSERT INTO %s (nullable_col) SELECT nationkey FROM nation", table.getName()), errorMessageForInsertIntoNotNullColumn("not_null_col"));
+            // TODO (https://github.com/trinodb/trino/issues/13551) This doesn't fail for other connectors so
+            //  probably shouldn't fail for MariaDB either. Once fixed, remove test override.
+            assertQueryFails(format("INSERT INTO %s (nullable_col) SELECT nationkey FROM nation WHERE regionkey < 0", table.getName()), ".*Field 'not_null_col' doesn't have a default value.*");
```

Comment:
There seems to be a problem here caused by the current implementation


I modified the test to run like this:
```
    public void testInsertIntoNotNullColumn()
    {
        try (TestTable table = new TestTable(getQueryRunner()::execute, "insert_not_null", "(nullable_col INTEGER, not_null_col INTEGER NOT NULL)")) {
            assertQuerySucceeds(format("INSERT INTO %s (nullable_col) SELECT nationkey FROM nation WHERE regionkey < 0", table.getName()));
        }
```

and I notice the following exception in the stacktrace:

```
java.lang.AssertionError: Expected query to succeed: INSERT INTO insert_not_null1evdhkhr9c (nullable_col) SELECT nationkey FROM nation WHERE regionkey < 0
...
    Suppressed: java.lang.Exception: SQL: INSERT INTO insert_not_null1evdhkhr9c (nullable_col) SELECT nationkey FROM nation WHERE regionkey < 0
.........
         Suppressed: java.lang.RuntimeException: Query: INSERT INTO `tpch`.`insert_not_null1evdhkhr9c` (`nullable_col`) SELECT `nullable_col` FROM `tpch`.`tmp_trino_2dc66901e1fa421f9cf3064225e9df13
```


There seems to be some side-effect of your changes.

Reply:
I get it now. The exception originates from MariaDB which complaints while copying the content from the temporary table towards the target table.